packages/victory-core/src/victory-clip-container/victory-clip-container.js

```diff
@@ -84,6 +85,7 @@ export default class VictoryClipContainer extends React.Component {
   }
 
   renderClippedGroup(props, clipId) {
+    const userProps = UserProps.getSafeUserProps(props);
```

Comment:
I don't love that we have to explicitly call this here when we are also calling this function in the top-level component. However, there are props here that we can't pass into the base HTML component, so we do need some way of identifying which "extra" props are okay to pass in (using `...rest` creates DOM warnings). There is a bigger issue here, which is the issue where we are passing this props object around, and don't have visibility into what is in it.

Reply:
Reading through this PR really opened by eyes to how much "magic" is happening in Victory to get (the appropriate) props applied to children.

This whitelisting of props (`safeUserProps`) seems to be flexible enough, but I agree it leads to a lot of manual checks and only checks for known cases of "unsafe" props.

Are there ideas to handle this differently in a future major version?

Reply:
Yep, two ideas. First, I think we should take a bunch of the logic for providing prop defaults and getting derived values from the component's props and put it in a context wrapper or some sort of top-level state. That way, we have a single source of truth for accessing the original props, defaults, and derived values. Right now that's happening in many different places with functions like `getBaseProps`, `getCalculatedProps`, and all these individual render functions. 

Second, I want to add explicit types to these helper and be more explicit about what is being passed into them to make it all a little less mysterious.

Reply:
Top-level context was the first thing I was thinking of as well. It'd be easy enough to define an API for the props of the top level component and build a context value with that, or even maybe introduce something like a `VictoryProvider` or something where the user can do it themselves.

This is helpful, thanks!